### PR TITLE
Refactor token lists

### PR DESCRIPTION
**Pull request by antooni**
This PR will refactor the current logic of passing token lists, instead of one big list, there will be a list for canonical tokens (deployed on Ethereum) and per-project token list (External & Native tokens)

Resolves L2B-2017

- [x] add common fake token method
- [x] test filtering 
- [x] migrate db
- [x] test migration locally
- [x] debug the runtime
- [x]  local sync

### CODE REVIEW

**Review by mateuszradomski — CHANGES_REQUESTED** [review]
Either I missed it or we do not have a test that tests if all tokens defined in token.json are CBV?

What about this point `Rename tokenList to canonicalTokensList`?

Are these assumptions meant to be implemented later?
```
// assumptions:
// AssetId needs to be unique, add test for it
// type TokenType = 'Canonical' | 'External' | 'Native'
// type ReportType = TokenType | 'Total'
```

File: packages/backend/src/core/reports/getEBVConfigHash.ts [hunk 1]
```diff
@@ -1,38 +1,19 @@
-import { hashJson } from '@l2beat/shared-pure'
-
-import { EBVToken } from '../assets'
-import { ReportProject } from './ReportProject'
+import { hashJson, Token } from '@l2beat/shared-pure'
 
 const EBV_LOGIC_VERSION = 0
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/backend/src/core/reports/getEBVConfigHash.ts:3]: no, I have migrated the USDC EBV to the new asset_id

File: packages/backend/src/core/totalSupply/getTotalSupplyConfigHash.ts [hunk 2]
```diff
@@ -1,27 +1,23 @@
-import { hashJson } from '@l2beat/shared-pure'
+import { hashJson, Token } from '@l2beat/shared-pure'
 import { sortBy } from 'lodash'
 
-import { TotalSupplyTokensConfig } from './TotalSupplyTokensConfig'
-
 // Increment this value to change the hash which in turn causes the system to
 // recalculate total supplies
 const TOTAL_SUPPLY_LOGIC_VERSION = 0
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/backend/src/core/totalSupply/getTotalSupplyConfigHash.ts:6]: answered above

**Comment by antooni** [comment]
> Either I missed it or we do not have a test that tests if all tokens defined in token.json are CBV?
> 
> What about this point `Rename tokenList to canonicalTokensList`?
> 
> Are these assumptions meant to be implemented later?
> 
> ```
> // assumptions:
> // AssetId needs to be unique, add test for it
> // type TokenType = 'Canonical' | 'External' | 'Native'
> // type ReportType = TokenType | 'Total'
> ```

For the ValueType refactor the linear task has been created
tokenList is exported as such (because it has all the tokens inside), for canonical there is a method getCanonicalTokens()
thanks for the AssetId test, i have forgot about this one

File: packages/backend/src/core/assets/ArbitrumEBVUpdater.ts [hunk 3]
```diff
@@ -19,11 +21,11 @@ import { PriceUpdater } from '../PriceUpdater'
 import { TaskQueue } from '../queue/TaskQueue'
 import { createEBVReports } from '../reports/createEBVReports'
 import { getEBVConfigHash } from '../reports/getEBVConfigHash'
-import { ReportProject } from '../reports/ReportProject'
 import { TotalSupplyUpdater } from '../totalSupply/TotalSupplyUpdater'
 import { AssetUpdater } from './AssetUpdater'
-import { EBVToken } from './EBVToken'
 
+// TODO: Make this class more generic
```
[the remark this answers is not in the document]
Comment (antooni) [reply packages/backend/src/core/assets/ArbitrumEBVUpdater.ts:27]: yup

**Review by mateuszradomski — APPROVED** [review]
I've synced locally and fixed an issue that arose, read the commit message from [a79bce2](https://github.com/l2beat/l2beat/pull/1771/commits/a79bce2f7652604e64b5cb7c8d25d2fe5ca02767) if you want to know more.

From my point of view this is ready to be merged, waiting for a second hand opinion

File: packages/backend/src/model/Project.ts [hunk 4]
```diff
@@ -2,37 +2,41 @@ import {
...
     projectId: layer2.id,
     type: 'layer2',
-    externalTokens: layer2.config.externalAssets ?? undefined,
     escrows: layer2.config.escrows.map((escrow) => ({
       address: escrow.address,
       sinceTimestamp: escrow.sinceTimestamp,
       tokens:
-        escrow.tokens === '*' ? tokenList : escrow.tokens.map(getTokenBySymbol),
+        escrow.tokens === '*'
+          ? tokenList.filter((t) => t.type === ValueType.CBV)
```
[the remark this answers is not in the document]
Comment (sz-piotr) [reply packages/backend/src/model/Project.ts:38]: We should probably just export `canonicalTokenList`

**Review by sz-piotr — APPROVED** [review]

**Pull request merged**